readme.md

```diff
@@ -14,121 +14,57 @@ $ npm install --save decompress
 ## Usage
 
 ```js
-const Decompress = require('decompress');
+const decompress = require('decompress');
 
-new Decompress({mode: '755'})
-	.src('foo.zip')
-	.dest('dest')
-	.use(Decompress.zip({strip: 1}))
-	.run();
+decompress('unicorn.zip', 'dist').then(files => {
+	console.log('done!');
+});
 ```
 
 
 ## API
 
-### new Decompress(options)
+### decompress(input, [output], [options])
 
-Creates a new `Decompress` instance.
-
-#### options.mode
-
-Type: `string`
-
-Set mode on the extracted files, i.e `{ mode: '755' }`.
-
-#### options.strip
-
-Type: `number`
-
-Equivalent to `--strip-components` for tar.
-
-### .src(files)
-
-#### files
-
-Type: `array`, `buffer` or `string`
-
-Set the files to be extracted.
-
-### .dest(path)
-
-#### path
-
-Type: `string`
-
-Set the destination to where your file will be extracted to.
-
-### .use(plugin)
-
-#### plugin
-
-Type: `function`
-
-Add a `plugin` to the middleware stack.
-
-### .run(callback)
-
-Extract your file with the given settings.
-
-#### callback(err, files)
-
-Type: `function`
-
-The callback will return an array of vinyl files in `files`.
-
-
-## Plugins
-
-The following [plugins](https://www.npmjs.org/browse/keyword/decompressplugin) are bundled with decompress:
-
-* [tar](#tar) — Extract TAR files.
-* [tar.bz2](#tarbz2) — Extract TAR.BZ files.
-* [tar.gz](#targz) — Extract TAR.GZ files.
-* [zip](#zip) — Extract ZIP files.
-
-### .tar(options)
-
-Extract TAR files.
+Returns a Promise for an array of files in the following format:
 
 ```js
-const Decompress = require('decompress');
-
-new Decompress()
-	.use(Decompress.tar({strip: 1}));
+{
+	data: Buffer,
+	mode: Number,
+	mtime: String,
+	path: String,
+	type: String
+}
 ```
 
-### .tarbz2(options)
+#### input
 
-Extract TAR.BZ files.
+Type: `string` `Buffer`
 
-```js
-const Decompress = require('decompress');
+File to decompress.
 
-new Decompress()
-	.use(Decompress.tarbz2({strip: 1}));
-```
+#### output
+
+Type: `string`
 
-### .targz(options)
+Output directory.
 
-Extract TAR.GZ files.
+#### options
 
-```js
-const Decompress = require('decompress');
+##### plugins
 
-new Decompress()
-	.use(Decompress.targz({strip: 1}));
-```
+Type: `array`<br>
+Default: `[decompressTar(), decompressTarGz(), decompressUnzip()]`
```

Comment:
Fixed in https://github.com/kevva/decompress/pull/38/commits/644d18f1690dd37649f6f92c440eb147741b3eb9.